### ISSUE TITLE
Player.EventListener eventListener is called twice in the BigPlayerFragment

### ISSUE THREAD

**Issue by shikha7763** [eneim/toro#288]
I am implementing eventListener in the BigPlayerFragment but the eventListener  is called twice.

**Comment by eneim** [comment]
@shikha7763 Can you show the part of code and the evident of the calling (logcat should be expected). Thanks.

**Comment by shikha7763** [comment]
@Override
    public void onStart() {
        super.onStart();
        if (playerHelper == null) {
            playerHelper = new ExoPlayerHelper(playerView,  //
                    DefaultRenderersFactory.EXTENSION_RENDERER_MODE_OFF, false);
            try {
                MediaSourceBuilder sourceBuilder = new MediaSourceBuilder(getContext(), Uri.parse(videoItem.getHlsurl()));
                playerHelper.prepare(sourceBuilder);
            } catch (ParserException e) {
                e.printStackTrace();
            }
        }
        playerHelper.setPlaybackInfo(playbackInfo);
        playerHelper.addEventListener(listener); //Here i am calling the eventlistener
        playerHelper.play();
    }

    private Handler handler = new Handler();
    private ExoPlayerHelper.EventListener listener = new ExoPlayerHelper.EventListener() {
        @Override
        public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
            super.onPlayerStateChanged(playWhenReady, playbackState);
            if (playbackState == Player.STATE_BUFFERING) {
                loading.setVisibility(View.VISIBLE);
                Log.v("TAG", "STATE_BUFFERING");
            } else if (playbackState == Player.STATE_ENDED) {
                handler.postDelayed(() -> {
                    int videoIndex = videoOrder;
                    if (videoIndex == newVideoItemsArrayList.size() - 1) {
                        Log.v("TAG", String.valueOf(videoIndex));
                        videoItem = newVideoItemsArrayList.get(videoIndex);
                        videoOrder = videoIndex;
                    } else {
                        Log.v("TAG", String.valueOf(videoIndex + 1));
                        videoItem = newVideoItemsArrayList.get(videoIndex + 1);
                        play(videoItem.getHlsurl());
                        videoOrder = videoIndex + 1;
                    }
                }, 300);
            } else {
                loading.setVisibility(View.GONE);
            }
        }
    };
In the log when the code executes the Log print the value twice in the BigPlayerFragment.

**Comment by shikha7763** [comment]
How can we implement on continues video playing in the BigPlayerFragment  and then get to the same position in the video list.

**Comment by eneim** [comment]
@shikha7763 Do you mean the code ``Log.v("TAG", "STATE_BUFFERING");`` shown twice every time in logcat?

Can you explain more about the second question: **continues video playing in the BigPlayerFragment** <-- do you mean to play many Videos by using one BigPlayerFragment?

**Comment by shikha7763** [comment]
yes the log show Log.v("TAG", "STATE_BUFFERING"); and I want when one video completes the video and play next video in the list and on changing orientation list must be on the same position in the BigPlayerFragment.

**Comment by eneim** [comment]
@shikha7763 About the double event, let me investigate.

About the second one, it requires more implementation steps that current demo would not cover. Let me think about improve this. The supporting for many Video is easy part (using ConcatinatedMediaSource) but to obtain the data and pass it back when User close BigPlayerFragment is the non-trivial part.

**Comment by shikha7763** [comment]
hey can you explain the process how can i implement the fullscreen feature on click listener.

**Comment by eneim** [comment]
@shikha7763 Oh forgot this question.

There are 2 kinds of "full screen": (1) Always landscape full-screen, (2) Both landscape and portrait full-screen.

In either kinds, the flow is:

**<user action> --> <prepare media data> --> <open full-screen player>**

So what you need to know in order to implement the behavior is (I will use question/answer base to explain):

- How would you like your Full-screen player be? --> You create the UI for it.
- What is the minimum data required to start a playback in your Full-screen player? --> You implement the mechanism to pass that data from "somewhere" to your Full-screen player. Normally it maybe via Intent or Database or whatever you are experienced with. 

This mechanism may or may not be tight to the implementation of your Full-screen player. For example, if you are using an Activity as Full-screen player, you have to use Intent (even if you pull data from Database, the ID of that data must be passed to the Activity for the query); but if you are using DialogFragment, you can use other mechanism like interface to communicate with the host and get data from it.

In my case, I use DialogFragment, but with Intent.

-  How the playback in Full-screen player should be? (or: can User interact to playback --> has Controller or not; has Ads or not --> need Ads support or not; auto recover on Config change or not --> need save/restore the state or not; etc).

- From where this Full-screen can be opened? --> Saying you have a static method to open the Full-screen player, here you need to provide this method to places where you want your User can open Full-screen player. It can be a Dialog, an Activity or whatever. And by providing the method, you also need to implement the part of **<prepare media data>**.

- How will your User open the Full-screen player? --> Here you need to intercept user action, by using Click listener or RxJava or whatever. The purpose is: by a specific action by User, you prepare the media data, call the static method to open your Full-screen player.

----

Above is the way of think when implementing the Full-screen player. If you have no question about it, I will go forward about implementation detail (that I used in my demo app).
